constraint-solver/src/solver.rs

```diff
@@ -222,18 +288,32 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display, BusInter: BusInterac
         }
     }
 
-    fn apply_assignment(&mut self, variable: &V, expr: &QuadraticSymbolicExpression<T, V>) -> bool {
+    fn apply_assignment(
+        &mut self,
+        variable: &Variable<V>,
+        expr: &QuadraticSymbolicExpression<T, Variable<V>>,
+    ) -> bool {
         log::debug!("({variable} := {expr})");
         self.constraint_system.substitute_by_unknown(variable, expr);
         self.assignments.push((variable.clone(), expr.clone()));
+
+        if let Variable::BusInteractionArg(..) = variable {
+            // TODO: Can there be more complex expressions here?
```

Comment:
Generally I'd assume yes, but looks like it doesn't happen right now? Should test with reth though